eb

```diff
@@ -60,71 +60,69 @@ PYTHON=
+    verbose "Considering '${python_cmd}'..."
 
     # check whether python* command being considered is available
     # (using 'command -v', since 'which' implies an extra dependency)
-    command -v $python_cmd &> /dev/null
-    if [ $? -eq 0 ]; then
+    if { command -v "${python_cmd}" && "${python_cmd}" -V; } &> /dev/null; then
 
         # make sure Python version being used is compatible
-        pyver=`$python_cmd -V 2>&1 | cut -f2 -d' '`
-        pyver_maj=`echo $pyver | cut -f1 -d'.'`
-        pyver_min=`echo $pyver | cut -f2 -d'.'`
-
-        if [ $pyver_maj -eq 2 ] && [ $pyver_min -ge $REQ_MIN_PY2VER ]; then
-            verbose "'$python_cmd' version: $pyver, which matches Python 2 version requirement (>= 2.$REQ_MIN_PY2VER)"
-            PYTHON=$python_cmd
-        elif [ $pyver_maj -eq 3 ] && [ $pyver_min -ge $REQ_MIN_PY3VER ]; then
-            verbose "'$python_cmd' version: $pyver, which matches Python 3 version requirement (>= 3.$REQ_MIN_PY3VER)"
-            PYTHON=$python_cmd
+        pyver=$("${python_cmd}" -V 2>&1 | cut -f2 -d' ')
+        pyver_maj=$(echo "${pyver}" | cut -f1 -d'.')
+        pyver_min=$(echo "${pyver}" | cut -f2 -d'.')
+
+        if [ "${pyver_maj}" -eq 2 ] && [ "${pyver_min}" -ge "${REQ_MIN_PY2VER}" ]; then
+            verbose "'${python_cmd}' version: ${pyver}, which matches Python 2 version requirement (>= 2.${REQ_MIN_PY2VER})"
+            PYTHON="${python_cmd}"
+        elif [ "${pyver_maj}" -eq 3 ] && [ "${pyver_min}" -ge "${REQ_MIN_PY3VER}" ]; then
+            verbose "'${python_cmd}' version: ${pyver}, which matches Python 3 version requirement (>= 3.${REQ_MIN_PY3VER})"
+            PYTHON="${python_cmd}"
         fi
 
-        if [ ! -z $PYTHON ]; then
+        if [ -n "${PYTHON}" ]; then
             # check whether EasyBuild framework is available for selected python command
-            $PYTHON -c "import $EASYBUILD_IMPORT_TEST" 2> /dev/null
-            if [ $? -eq 0 ]; then
-                verbose "'$python_cmd' is able to import '$EASYBUILD_IMPORT_TEST', so retaining it"
+            if "${PYTHON}" -c "import ${EASYBUILD_IMPORT_TEST}" 2> /dev/null; then
+                verbose "'${python_cmd}' is able to import '${EASYBUILD_IMPORT_TEST}', so retaining it"
             else
                 # if EasyBuild framework is not available, don't use this python command, keep searching...
-                verbose "'$python_cmd' is NOT able to import '$EASYBUILD_IMPORT_TEST' so NOT retaining it"
+                verbose "'${python_cmd}' is NOT able to import '${EASYBUILD_IMPORT_TEST}' so NOT retaining it"
                 unset PYTHON
             fi
         fi
 
         # break out of for loop if we've found a valid python command
-        if [ ! -z $PYTHON ]; then
+        if [ -n "${PYTHON}" ]; then
             break
         fi
     else
-        verbose "No '$python_cmd' found in \$PATH, skipping..."
+        verbose "No valid '${python_cmd}' found in \$PATH, skipping..."
```

Comment:
@Poshi Why the "`valid`" here, can you clarify?

This message will be printed when `command -v ${python_cmd}` fails, or if `${python_cmd} -V` fails, which pretty much means there's no working `${python_cmd}`. Maybe it should say "`working`" rather than "`valid`"?